src/main/java/com/invoiced/entity/AbstractEntity.java

```diff
@@ -273,9 +312,32 @@ public void delete() throws EntityException {
 			return;
 		}
 
+		String url = this.getEndpoint(true, true);
+
+
+		try {
+
+			this.conn.delete(url);
+
+		} catch (Throwable c) {
+
+			throw new EntityException(c);
+		}
+	}
+
+	public void delete(boolean includeId) throws EntityException {
+
+		if (!this.hasCRUD()) {
+			return;
```

Comment:
I see this pattern in a few places. When this happens it means that a method is being called mistakenly, likely due to some mistake on the developer's part. I think it's better to throw an exception in these situations instead of silently returning. This could lead to situations where a caller thinks the operation happened when it actually did not.

Reply:
Agreed and changes made.